src/Algebra/Graph/Relation/InternalDerived.hs

```diff
@@ -14,16 +14,19 @@
 -----------------------------------------------------------------------------
 module Algebra.Graph.Relation.InternalDerived (
```

Comment:
I think you can move all code related to symmetric relations into a separate module `Symmetric.Internal`.